src/baldr/graphtile.cc

```diff
@@ -273,7 +273,10 @@ void GraphTile::Initialize(const GraphId& graphid) {
                              " vs raw tile data size = " + std::to_string(tile_size) +
                              ". Tile file might me corrupted");
 
-  // TODO check version
+  if (int current_version = header_->version()[0] - '0'; current_version != VALHALLA_VERSION_MAJOR) {
```

Comment:
warn if the tiles were built with another major version of valhalla. could even throw, it's almost guaranteed that it won't be backwards compatible.